cobra/model_building/forward_selection.py

```diff
@@ -253,25 +248,36 @@ def _find_next_best_model(self, train_data: pd.DataFrame,
         """
         # placeholders
         best_model = None
-        best_performance = -1
+        if self.MLModel == LogisticRegressionModel:
+            best_performance = -1  # AUC metric is used.
+        elif self.MLModel == LinearRegressionModel:
+            best_performance = float("inf")  # RMSE metric is used.
+        else:
+            raise ValueError("No metric comparison method has been configured "
+                             "for the given model_type specified as "
+                             "ForwardFeatureSelection argument.")
 
         for pred in candidate_predictors:
-
-            # train model with additional predictor
+            # Train a model with an additional predictor
             model = self._train_model(train_data, target_column_name,
                                       (current_predictors + [pred]))
-            # Evaluate model
+            # Evaluate the model
             performance = (model
                            .evaluate(train_data[current_predictors + [pred]],
                                      train_data[target_column_name],
                                      split="train"))
 
-            if (self.pos_only and (not (model.get_coef() >= 0).all())):
+            if self.pos_only and (not (model.get_coef() >= 0).all()):
                 continue
 
-            # check if model is better than current best model
-            # and if yes, replace current best!
-            if (performance >= best_performance):
```

Comment:
changed the comparison logic from performance >= best_performance to performance > best_performance (regression example): to me it seems adding a next best model is pointless if it just performs as good as the previous selected feature... (or do I miss something?)